roles/acme_letsencrypt/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+---
+- name: preconditions
+  include_tasks: preconditions.yml
+
+- name: run key generation
+  include_tasks: create-keys.yml
+
+- name: "select provider {{ acme_letsencrypt_challenge_provider }}"
```

Comment:
If we put this task inside the precondition, there are only include-tasks in the `main.yml`. This makes the structure easier to reason about, since we don't have to think if there are tasks hidden beetween the includes.

Reply:
Good idea but i would keep the include and just move it to the preconditions.

Reply:
Moved the "select provider" task into preconditions